gamemodes/terrortown/gamemode/shared/sh_init.lua

```diff
@@ -663,20 +663,9 @@ function EquipmentIsBuyable(tbl, ply)
 	end
 
 	if tbl.minPlayers and tbl.minPlayers > 1 then
-		local choices = {}
-		local plys = player.GetAll()
-
-		for i = 1, #plys do
-			local v = plys[i]
-
-			-- everyone on the forcespec team is in specmode
-			if not IsValid(v) or v:GetForceSpec() then continue end
-
-			choices[#choices + 1] = v
-		end
-
-		if #choices < tbl.minPlayers then
-			return false, " " .. #choices .. " / " .. tbl.minPlayers, "Minimum amount of active players needed."
+		local activePlayers = GetGlobalInt("ttt2_active_players", 0)
```

Comment:
I'd much prefer having a shared getter method for this `GlobalInt`. This greatly improves the documentation

Reply:
> I'd much prefer having a shared getter method for this `GlobalInt`. This greatly improves the documentation

Sure, where should I put them? @Alf21, could you tell me (us) where I should best put the getter and setter function?

Reply:
This won't work at all. The client should never know the exact amount of alive players. So a shared global doesn't make sense. Instead, just the available equipment should be sent by the server. As you see, the problem is a way more complex.

Reply:
Additionally, I don't see any issue here. Players that aren't alive but forced spectators were alive at some time in the round. As a result, the equip should be available, because the requirements have been filled.

Reply:
The problem without any changes is, that even players that join during a around count as players. I though about using WasActiveInRound, but this will lead to the item not being buyable if somebody leaves and the minPlayers amount is not fullfiled anymore. Also, I mean a player does already "know" how many players are alive? Pretty much because he knows the alive state.

Reply:
Yea you are right, sad that `Alive` is a shared function for a multiplayer game... Anyways, I meant the combination with `IsTerror`. Wouldn't a clientside combination of `Alive() and GetForceSpec()` doing the trick? If not, then it's an issue with Alive() which probably should get patched

Reply:
A combination would still not work as it should, as it would still not count players that disconnect from the Server, which would be pretty unfair for Traitors suddenly not being able to buy an item anymore, just because somebody left.

Reply:
Caching the exact amount of active players at the beginning of the round should do the trick.

Reply:
Well, I do this in this PR xD?

Reply:
Yea, but then there still isn't any need for a shared global var (the issue we are discussing about here).

Reply:
But EquipmentIsBuyable is called on the client aswell, so it would not grey out the Item, but just give out an error when trying to buy it.

Reply:
Then we have to modify it. There is no need of adding an additional networked var if the client is already aware of the attributes needed for further "calculations".

Reply:
Okay well.. This is getting a bit over my head rn. I do understand, but I do not understand how exactly you'd like to have it.
Sure, not GlobalInt, alright. You said caching, but this also needs to be done in client and Server. Soo? Should it be a global variabel which is shared?

Reply:
Yea, a shared var that isn't synced (because we don't need a sync if the client and the server are already aware of the attributes needed to get to the same state).

Reply:
Okay well.. I could now go over "but what if a client joins after round begun, and gets respawned? He wont know the start player count" thingy.. But if thats okay, sure.. I just need to know, where I should initialize the variable, and how to call it.